### DISCUSSION TITLE
Rule to compare with months ago

### DISCUSSION THREAD

**Discussion by andre-picoli-ext-dock** [zeroSteiner/rule-engine#53] [Q&A] [answered]
How can I compare the current date with 3 months ago using python rule_engine?
I need to compare if establishment date is before 3 months ago, changing the rule.

Here's the code I'm trying to solve:
```
import rule_engine
from datetime import datetime


company = {
    'name': 'Company name',
    "establishmentDate": "2021-04-08"
  }
rule = rule_engine.Rule(
  '("today()" - "3 months" > establishmentDate)'
)

rule.matches(company)
```

**Comment by zeroSteiner** [comment, marked as answer]
You would use the [TIMEDELTA](https://zerosteiner.github.io/rule-engine/types.html#datetime-and-timedelta) data type that was added in v3.5. It doesn't support months exactly because months do not always have the same number of days. You can, however, check if the date is older than 90 days using `($today - t'P90D') > establishmentDate`.

Example:
```
python -m rule_engine.debug_repl --edit-console
edit the 'context' and 'thing' objects as necessary
>>> import datetime
>>> thing = dict(name='Company Name', establishmentDate=datetime.date(2021, 4, 8))
>>> 
exiting the edit console...
rule > establishmentDate
result: 
datetime.datetime(2021, 4, 8, 0, 0, tzinfo=tzlocal())
rule > $today
result: 
datetime.datetime(2023, 2, 7, 0, 0, tzinfo=tzlocal())
rule > $today - t'P90D'
result: 
datetime.datetime(2022, 11, 9, 0, 0, tzinfo=tzlocal())
rule > ($today - t'P90D') > establishmentDate
result: 
True
rule > 
```

**Reply by andre-picoli-ext-dock** [reply]
Thanks @zeroSteiner! The solution was perfect.
